### PR TITLE
find cpu_amp is incorrect set, it's only set if self.sharded_ddp is n…

### PR DESCRIPTION
**Pull request by sywangyi**
…ot None...

- trainer: @sgugger

### CODE REVIEW

**Comment by sywangyi** [comment]
amp does not work if self.sharded_ddp is None

**Review by sgugger — COMMENTED** [review]
cc @pacman100

File: src/transformers/trainer.py [hunk 1]
```diff
@@ -635,8 +635,7 @@ def __init__(
                         f"FP16 provided in SM_HP_MP_PARAMETERS is {smp.state.cfg.fp16}, "
                         "but SageMaker Model Parallelism < 1.10 does not support FP16 in trainer."
                     )
-
-        if (args.fp16 or args.bf16) and self.sharded_ddp is not None:
+        if args.fp16 or args.bf16:
```
Comment (pacman100) [inline src/transformers/trainer.py:638]: Now, `mixed precision` handling is moved to Accelerate. Only when Fairscale is being used mixed precision handling is supported in trainer as it isn't present in Accelerate.

Comment (pacman100) [reply src/transformers/trainer.py:638]: So, this change can't be done

File: src/transformers/trainer.py [hunk 2]
```diff
@@ -653,30 +652,29 @@ def __init__(
...
-                    #  bf16 does not need grad scaling
-                    self.do_grad_scaling = self.amp_dtype == torch.float16
-                    if self.do_grad_scaling:
-                        if self.sharded_ddp is not None:
-                            self.scaler = ShardedGradScaler()
-                        elif self.fsdp is not None:
-                            from torch.distributed.fsdp.sharded_grad_scaler import (
-                                ShardedGradScaler as FSDPShardedGradScaler,
-                            )
+            if args.half_precision_backend == "cuda_amp":
```
Comment (pacman100) [inline src/transformers/trainer.py:655]: same reason as above

**Review by pacman100 — COMMENTED** [review]
Now, mixed precision handling is moved to Accelerate. Only when Fairscale is being used mixed precision handling is supported in trainer as it isn't present in Accelerate.

**Comment by pacman100** [comment]
> Hi, I see in evaluation_loop and prediction. accelerate prepare is only called when self.is_deepspeed_enabled and self.model_wrapped is self.model, so how about the non-deepspeed case?

That makes sense, thank you! I've raised a PR addressing this as linked above.

**Comment by sywangyi** [comment]
Hi, @pacman100 , I have another question. any idea of jit trace in evaluation in https://github.com/huggingface/transformers/blob/main/src/transformers/trainer.py#L1384, it also need the amp info.

**Pull request closed without merging**